packages/core/test/tag/compoundTagTests.tsx

```diff
@@ -33,13 +34,24 @@ describe("<CompoundTag>", () => {
 
     it("renders icons", () => {
         const wrapper = shallow(
-            <CompoundTag icon="tick" rightIcon="airplane" leftContent="Hello">
+            <CompoundTag icon="tick" endIcon="airplane" leftContent="Hello">
                 World
             </CompoundTag>,
         );
         assert.lengthOf(wrapper.find(Icon), 2);
     });
 
+    it("prefers endIcon to rightIcon", () => {
+        const result = render(
+            // eslint-disable-next-line @typescript-eslint/no-deprecated
+            <CompoundTag endIcon="airplane" rightIcon="tick" leftContent="Hello">
+                World
+            </CompoundTag>,
+        );
+        expect(result.baseElement.querySelector(`.${Classes.ICON}-airplane`)).to.exist;
+        expect(result.baseElement.querySelector(`.${Classes.ICON}-add`)).not.to.exist;
+    });
```

Comment:
Also, it might be slightly more robust/clean to opt for testids here vs. querying the dom structure. You'll be able to use some of RTL's nicer methods, and it will test fewer implementation details (such as the structure/name of the class names)

```tsx
it("prefers endIcon to rightIcon", () => {
    const endIcon = <Icon icon="airplane" data-testid="endIcon" />;
    const rightIcon = <Icon icon="tick" data-testid="rightIcon" />;
    render(
        // eslint-disable-next-line @typescript-eslint/no-deprecated
        <CompoundTag endIcon={endIcon} rightIcon={rightIcon} leftContent="Hello">
            World
        </CompoundTag>,
    );

    expect(screen.getByTestId("endIcon")).to.exist;
    expect(screen.queryByTestId("rightIcon")).not.to.exist;
});
```